File: rust/src/telemetry.rs
use inapi::{Host, Telemetry};
use std::process::Command;
use super::Testable;

pub struct TelemetryTest;

impl Testable for TelemetryTest {
    fn test(mut host: &mut Host) {
        let telemetry = Telemetry::init(&mut host).unwrap();
        let output = Command::new("hostname").output().unwrap();
        assert!(output.status.success());
        assert_eq!(String::from_utf8(output.stdout).unwrap(), telemetry.hostname);
    }
}

File: php/src/telemetry.php
<?php

use Intecture\Telemetry;

class TelemetryTest implements Testable {
    public static function test($host) {
        $telemetry = Telemetry::load($host);
        $tdata = $telemetry->get();
        $out = $exit = NULL;
        $check_start = exec('hostname -f', $out, $exit);
        assert($exit == 0);
        assert($tdata['hostname'] == $out[0]);
    }
}
